plugins/inputs/wireless/README.md

```diff
@@ -9,30 +9,69 @@ The wireless plugin gathers metrics about wireless link quality by reading the `
 [[inputs.wireless]]
   ## Sets 'proc' directory path
   ## If not specified, then default is /proc
+  ## Ignored on macOS/darwin
   # host_proc = "/proc"
 ```
 
+The `host_proc` option is ignored on macOS.
+
 ## Metrics
 
+### Linux metrics
+
 - metric
   - tags:
     - interface (wireless interface)
   - fields:
-    - status (int64, gauge) - Its current state. This is a device dependent information
-    - link (int64, percentage, gauge) - general quality of the reception
-    - level (int64, dBm, gauge) - signal strength at the receiver
-    - noise (int64, dBm, gauge) - silence level (no packet) at the receiver
-    - nwid (int64, packets, counter) - number of discarded packets due to invalid network id
-    - crypt (int64, packets, counter) - number of packet unable to decrypt
-    - frag (int64, packets, counter) - fragmented packets
-    - retry (int64, packets, counter) - cumulative retry counts
-    - misc (int64, packets, counter) - dropped for un-specified reason
-    - missed_beacon (int64, packets, counter) - missed beacon packets
+    - status (int64, metric) - Its current state. This is a device dependent information
+    - link (int64, percentage, metric) - general quality of the reception
+    - level (int64, dBm, metric) - signal strength at the receiver
+    - noise (int64, dBm, metric) - silence level (no packet) at the receiver
+    - nwid (int64, packets, metric) - number of discarded packets due to invalid network id
+    - crypt (int64, packets, metric) - number of packet unable to decrypt
+    - frag (int64, packets, metric) - fragmented packets
+    - retry (int64, packets, metric) - cumulative retry counts
+    - misc (int64, packets, metric) - dropped for un-specified reason
+    - missed_beacon (int64, packets, metric) - missed beacon packets
+
+### macOS metrics
+
+- metric
+  - tags:
+    - state (running or not)
+    - op_mode (operating mode: station, ad_hoc or ap)
+    - 802.11_auth (open or hidden)
+    - link_auth (authorization scheme)
 
+  - fields:
+    - BSSID (string, mac address, _only_ reported if run as `root`)
+    - SSID (string, The SSID of the network)
+    - agrCtlRSSI (int64, dBm, metric) - The current aggregate RSSI of the link
+    - agrExtRSSI (int64, dBm, metric) - The current aggregate external RSSI of the link
+    - agrCtlNoise (int64, dBm, metric) - The current aggregate noise of the link
+    - agrExtNoise (int64, dBm, metric) - The current aggregate external noise of the link
+    - lastTxRate (int64, Mbps, metric) - The last transmit rate
+    - maxRate (int64, Mbps, metric) - The maximum transmit rate
+    - lastAssocStatus (int64, metric) - The last association status
+    - MCS (int64, MCS, metric) - The last MCS
+    - guardInterval (int64, guard, metric) - The guard interval
+    - NSS (int64, NSS, metric) - The number of spatial streams
+    - channel (channel information)
```

Comment:
It is much less useful for the plugin produce a completely different metric on mac. Users who want to monitor wireless performance on multiple computers including mac and linux would have to use two different names for the same thing. For example, the linux metric's `level` field is a measure of signal strength, but `agrCtlRSSI` and `agrExtRSSI` are too and I wouldn't be surprised if one of them is the same as `level`. Similar for `noise` and `agrCtlNoise`/`agrExtNoise`

Could you use the same metric format as linux, or at least a subset of it? I understand in #10622 you started out with a separate plugin and were encouraged to merge it with this plugin. If it's not possible to use the same metric format, maybe it would be better to add mac support as a separate plugin.

Reply:
See comment above.